CC related crash when building  `scala-library` docs

## Compiler version
3.8.0-RC1

Scaladoc is crushing when building `scala-library` - it blocks #24429 and #24433 - Sonatype requires `javadoc` artifacts to be published as part of validation

## Reproduction steps 

```scala
scala-library-nonbootstrapped / Compile / doc
```
or in #24433 
```scala
scala-library-bootstrapped / Compile / doc
```

## Output (click arrow to expand)
<!--
Note: to get a full crash log when using scala-cli, run compilation with the flag `--server=false`
-->
<details>

```scala
[info] 
[info]   exception occurred while readTasty class file scala.collection.ArrayOps
[info] 
[info]   An unhandled exception was thrown in the compiler.
[info]   Please file a crash report here:
[info]   https://github.com/scala/scala3/issues/new/choose
[info]   For non-enriched exceptions, compile with -Xno-enrich-error-messages.
[info] 
[info] 
[info]      while compiling: class file scala.collection.ArrayOps
[info]         during phase: readTasty
[info]                 mode: Mode(ImplicitsEnabled,ReadPositions)
[info]      library version: version 2.13.16
[info]     compiler version: version 3.7.4
[info]             settings: -Xread-docs true -Yretain-trees true -classpath /Users/wmazur/projects/scala/scala3/library/target/scala-library-nonbootstrapped/classes/:dummy_c5638eb4:/Users/wmazur/.sbt/boot/scala-2.12.20/org.scala-sbt/sbt/1.11.5/compiler-interface-1.10.8.jar:/Users/wmazur/.sbt/boot/scala-2.12.20/org.scala-sbt/sbt/1.11.5/test-interface-1.0.jar:/Users/wmazur/.sbt/boot/scala-2.12.20/org.scala-sbt/sbt/1.11.5/util-interface-1.11.5.jar:/Users/wmazur/.sbt/boot/scala-2.12.20/org.scala-sbt/sbt/1.11.5/jline-terminal-3.27.1.jar:/Users/wmazur/.sbt/boot/scala-2.12.20/org.scala-sbt/sbt/1.11.5/jline-terminal-jni-3.27.1.jar:/Users/wmazur/.sbt/boot/scala-2.12.20/org.scala-sbt/sbt/1.11.5/jline-2.14.7-sbt-9a88bc413e2b34a4580c001c654d1a7f4f65bf18.jar:/Users/wmazur/.sbt/boot/scala-2.12.20/org.scala-sbt/sbt/1.11.5/jline-native-3.27.1.jar:/Users/wmazur/.sbt/boot/scala-2.12.20/org.scala-sbt/sbt/1.11.5/jansi-2.4.1.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-library_3/3.7.4/scala3-library_3-3.7.4.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.16/scala-library-2.13.16.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-library_3/3.7.4/scala3-library_3-3.7.4.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.16/scala-library-2.13.16.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-compiler_3/3.7.4/scala3-compiler_3-3.7.4.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-interfaces/3.7.4/scala3-interfaces-3.7.4.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/tasty-core_3/3.7.4/tasty-core_3-3.7.4.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-asm/9.8.0-scala-1/scala-asm-9.8.0-scala-1.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-sbt/compiler-interface/1.10.7/compiler-interface-1.10.7.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-reader/3.29.0/jline-reader-3.29.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-terminal/3.29.0/jline-terminal-3.29.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-terminal-jni/3.29.0/jline-terminal-jni-3.29.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-sbt/util-interface/1.10.7/util-interface-1.10.7.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-native/3.29.0/jline-native-3.29.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-compiler_3/3.7.4/scala3-compiler_3-3.7.4.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-interfaces/3.7.4/scala3-interfaces-3.7.4.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/tasty-core_3/3.7.4/tasty-core_3-3.7.4.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-asm/9.8.0-scala-1/scala-asm-9.8.0-scala-1.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-sbt/compiler-interface/1.10.7/compiler-interface-1.10.7.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-reader/3.29.0/jline-reader-3.29.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-terminal/3.29.0/jline-terminal-3.29.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-terminal-jni/3.29.0/jline-terminal-jni-3.29.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-sbt/util-interface/1.10.7/util-interface-1.10.7.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jline/jline-native/3.29.0/jline-native-3.29.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scaladoc_3/3.7.4/scaladoc_3-3.7.4.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-tasty-inspector_3/3.7.4/scala3-tasty-inspector_3-3.7.4.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark/0.62.2/flexmark-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util-ast/0.62.2/flexmark-util-ast-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util-data/0.62.2/flexmark-util-data-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util-html/0.62.2/flexmark-util-html-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-ext-anchorlink/0.62.2/flexmark-ext-anchorlink-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-ext-autolink/0.62.2/flexmark-ext-autolink-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-ext-emoji/0.62.2/flexmark-ext-emoji-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-ext-gfm-strikethrough/0.62.2/flexmark-ext-gfm-strikethrough-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-ext-gfm-tasklist/0.62.2/flexmark-ext-gfm-tasklist-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-ext-wikilink/0.62.2/flexmark-ext-wikilink-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-ext-tables/0.62.2/flexmark-ext-tables-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-ext-yaml-front-matter/0.62.2/flexmark-ext-yaml-front-matter-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/nl/big-o/liqp/0.8.2/liqp-0.8.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jsoup/jsoup/1.17.2/jsoup-1.17.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/dataformat/jackson-dataformat-yaml/2.15.1/jackson-dataformat-yaml-2.15.1.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util-builder/0.62.2/flexmark-util-builder-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util-collection/0.62.2/flexmark-util-collection-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util-dependency/0.62.2/flexmark-util-dependency-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util-format/0.62.2/flexmark-util-format-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util-misc/0.62.2/flexmark-util-misc-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util-sequence/0.62.2/flexmark-util-sequence-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util-visitor/0.62.2/flexmark-util-visitor-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/jetbrains/annotations/15.0/annotations-15.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util/0.62.2/flexmark-util-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/nibor/autolink/autolink/0.6.0/autolink-0.6.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-jira-converter/0.62.2/flexmark-jira-converter-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/antlr/antlr4-runtime/4.7.2/antlr4-runtime-4.7.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-annotations/2.15.1/jackson-annotations-2.15.1.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-core/2.15.1/jackson-core-2.15.1.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-databind/2.15.1/jackson-databind-2.15.1.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/fasterxml/jackson/datatype/jackson-datatype-jsr310/2.12.1/jackson-datatype-jsr310-2.12.1.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/ua/co/k/strftime4j/1.0.5/strftime4j-1.0.5.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/yaml/snakeyaml/2.0/snakeyaml-2.0.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-util-options/0.62.2/flexmark-util-options-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-ext-ins/0.62.2/flexmark-ext-ins-0.62.2.jar:/Users/wmazur/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/vladsch/flexmark/flexmark-ext-superscript/0.62.2/flexmark-ext-superscript-0.62.2.jar -d /Users/wmazur/projects/scala/scala3/library/target/scala-library-nonbootstrapped/api -deprecation true -encoding UTF8 -feature true -from-tasty true -java-output-version 17 -sourcepath /Users/wmazur/projects/scala/scala3/library/src:/Users/wmazur/projects/scala/scala3/library/src-non-bootstrapped:/Users/wmazur/projects/scala/scala3/library/target/scala-library-nonbootstrapped/src_managed/main
[error] java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
[error]         at xsbt.DottydocRunner.run(DottydocRunner.java:65)
[error]         at xsbt.ScaladocInterface.run(ScaladocInterface.java:11)
[error]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[error]         at sbt.internal.inc.AnalyzingCompiler.invoke(AnalyzingCompiler.scala:329)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:175)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:133)
[error]         at sbt.Doc$.$anonfun$scaladoc$1(Doc.scala:53)
[error]         at sbt.Doc$.$anonfun$scaladoc$1$adapted(Doc.scala:41)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1(RawCompileLike.scala:80)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1$adapted(RawCompileLike.scala:73)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4(RawCompileLike.scala:64)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4$adapted(RawCompileLike.scala:62)
[error]         at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:220)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1(RawCompileLike.scala:69)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1$adapted(RawCompileLike.scala:53)
[error]         at sbt.Defaults$.$anonfun$docTaskSettings$4(Defaults.scala:2276)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error]         at sbt.Execute.work(Execute.scala:292)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[error]         at java.base/java.lang.Thread.run(Thread.java:1583)
[error] Caused by: java.lang.reflect.InvocationTargetException
[error]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:109)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[error]         at xsbt.DottydocRunner.run(DottydocRunner.java:61)
[error]         at xsbt.ScaladocInterface.run(ScaladocInterface.java:11)
[error]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[error]         at sbt.internal.inc.AnalyzingCompiler.invoke(AnalyzingCompiler.scala:329)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:175)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:133)
[error]         at sbt.Doc$.$anonfun$scaladoc$1(Doc.scala:53)
[error]         at sbt.Doc$.$anonfun$scaladoc$1$adapted(Doc.scala:41)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1(RawCompileLike.scala:80)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1$adapted(RawCompileLike.scala:73)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4(RawCompileLike.scala:64)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4$adapted(RawCompileLike.scala:62)
[error]         at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:220)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1(RawCompileLike.scala:69)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1$adapted(RawCompileLike.scala:53)
[error]         at sbt.Defaults$.$anonfun$docTaskSettings$4(Defaults.scala:2276)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error]         at sbt.Execute.work(Execute.scala:292)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[error]         at java.base/java.lang.Thread.run(Thread.java:1583)
[error] Caused by: java.lang.ClassCastException: class dotty.tools.dotc.core.Phases$NoPhase$ cannot be cast to class dotty.tools.dotc.cc.CheckCaptures (dotty.tools.dotc.core.Phases$NoPhase$ and dotty.tools.dotc.cc.CheckCaptures are in unnamed module of loader sbt.internal.classpath.ClassLoaderCache$Key$CachedClassLoader @10466ac0)
[error]         at dotty.tools.dotc.cc.CaptureOps$package$.ccState(CaptureOps.scala:53)
[error]         at dotty.tools.dotc.cc.CaptureSet$VarState$.HardSeparate(CaptureSet.scala:1374)
[error]         at dotty.tools.dotc.cc.CaptureSet.$plus$plus(CaptureSet.scala:318)
[error]         at dotty.tools.dotc.cc.CaptureSet$.mapRefs$$anonfun$1(CaptureSet.scala:1222)
[error]         at dotty.tools.dotc.util.SimpleIdentitySet$Set1.$div$colon(SimpleIdentitySet.scala:101)
[error]         at dotty.tools.dotc.cc.CaptureSet$.mapRefs(CaptureSet.scala:1222)
[error]         at dotty.tools.dotc.cc.CaptureSet$.mapRefs(CaptureSet.scala:1226)
[error]         at dotty.tools.dotc.cc.CaptureSet.map(CaptureSet.scala:401)
[error]         at dotty.tools.dotc.core.Types$TypeMap.mapCapturingType(Types.scala:6294)
[error]         at dotty.tools.dotc.core.Types$TypeMap.mapOver(Types.scala:6389)
[error]         at dotty.tools.dotc.core.Types$FollowAliasesMap.mapFollowingAliases(Types.scala:6217)
[error]         at dotty.tools.dotc.core.Types$FollowAliasesMap.mapFollowingAliases$(Types.scala:6210)
[error]         at dotty.tools.dotc.core.TypeOps$$anon$5.mapFollowingAliases(TypeOps.scala:956)
[error]         at dotty.tools.dotc.core.TypeOps$$anon$5.apply(TypeOps.scala:963)
[error]         at dotty.tools.dotc.core.NamerOps$.suppressIntoIfParam(NamerOps.scala:371)
[error]         at dotty.tools.dotc.typer.Namer$Completer.typeSig(Namer.scala:838)
[error]         at dotty.tools.dotc.typer.Namer$Completer.completeInCreationContext(Namer.scala:1005)
[error]         at dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:874)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:175)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:190)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:192)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:403)
[error]         at dotty.tools.dotc.typer.Typer.retrieveSym(Typer.scala:3623)
[error]         at dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3648)
[error]         at dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3769)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3857)
[error]         at dotty.tools.dotc.typer.Typer.typed(Typer.scala:3862)
[error]         at dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3973)
[error]         at dotty.tools.dotc.typer.Namer.typedAheadExpr$$anonfun$1(Namer.scala:1802)
[error]         at dotty.tools.dotc.typer.Namer.typedAhead(Namer.scala:1792)
[error]         at dotty.tools.dotc.typer.Namer.typedAheadExpr(Namer.scala:1802)
[error]         at dotty.tools.dotc.typer.Namer.completeParams$1$$anonfun$1(Namer.scala:2012)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error]         at scala.collection.immutable.List.foreach(List.scala:334)
[error]         at dotty.tools.dotc.typer.Namer.completeParams$1(Namer.scala:2010)
[error]         at dotty.tools.dotc.typer.Namer.completeTrailingParamss$$anonfun$2(Namer.scala:2019)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error]         at scala.collection.immutable.List.foreach(List.scala:334)
[error]         at dotty.tools.dotc.typer.Namer.completeTrailingParamss(Namer.scala:2019)
[error]         at dotty.tools.dotc.typer.Namer.defDefSig(Namer.scala:1957)
[error]         at dotty.tools.dotc.typer.Namer$Completer.typeSig(Namer.scala:845)
[error]         at dotty.tools.dotc.typer.Namer$Completer.completeInCreationContext(Namer.scala:1005)
[error]         at dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:874)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:175)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:190)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:192)
[error]         at dotty.tools.dotc.core.Denotations$SingleDenotation.signature(Denotations.scala:620)
[error]         at dotty.tools.dotc.core.Denotations$SingleDenotation.signature(Denotations.scala:612)
[error]         at dotty.tools.dotc.core.Denotations$SingleDenotation.atSignature(Denotations.scala:653)
[error]         at dotty.tools.dotc.core.Denotations$SingleDenotation.atSignature(Denotations.scala:649)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler.dotty$tools$dotc$core$tasty$TreeUnpickler$TreeReader$$_$_$searchDenot$1(TreeUnpickler.scala:1587)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readLengthTree$1(TreeUnpickler.scala:1591)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readTree(TreeUnpickler.scala:1693)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readLengthTree$1(TreeUnpickler.scala:1492)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readTree(TreeUnpickler.scala:1693)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readLengthTree$1(TreeUnpickler.scala:1484)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readTree(TreeUnpickler.scala:1693)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler.readRhs$1$$anonfun$1$$anonfun$1(TreeUnpickler.scala:908)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$LazyReader.complete(TreeUnpickler.scala:1833)
[error]         at dotty.tools.dotc.ast.Trees$WithLazyFields.force(Trees.scala:1134)
[error]         at dotty.tools.dotc.ast.Trees$WithLazyFields.force$(Trees.scala:1130)
[error]         at dotty.tools.dotc.ast.Trees$ValOrDefDef.force(Trees.scala:417)
[error]         at dotty.tools.dotc.ast.Trees$DefDef.forceFields(Trees.scala:941)
[error]         at dotty.tools.dotc.ast.Trees$DefDef.rhs(Trees.scala:942)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1759)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1808)
[error]         at dotty.tools.dotc.CompilationUnit$Force.traverse(CompilationUnit.scala:213)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1807)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1807)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.fold$1(Trees.scala:1671)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.apply(Trees.scala:1673)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1766)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1808)
[error]         at dotty.tools.dotc.CompilationUnit$Force.traverse(CompilationUnit.scala:213)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1807)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1807)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1763)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1808)
[error]         at dotty.tools.dotc.CompilationUnit$Force.traverse(CompilationUnit.scala:213)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1807)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1807)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.fold$1(Trees.scala:1671)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.apply(Trees.scala:1673)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1772)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1808)
[error]         at dotty.tools.dotc.CompilationUnit$Force.traverse(CompilationUnit.scala:213)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1807)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1807)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.fold$1(Trees.scala:1671)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.apply(Trees.scala:1673)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1772)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1677)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1808)
[error]         at dotty.tools.dotc.CompilationUnit$Force.traverse(CompilationUnit.scala:213)
[error]         at dotty.tools.dotc.CompilationUnit$.apply(CompilationUnit.scala:153)
[error]         at dotty.tools.dotc.CompilationUnit$.apply(CompilationUnit.scala:144)
[error]         at dotty.tools.dotc.fromtasty.ReadTasty.compilationUnit$1(ReadTasty.scala:57)
[error]         at dotty.tools.dotc.fromtasty.ReadTasty.readTASTY(ReadTasty.scala:85)
[error]         at dotty.tools.dotc.fromtasty.ReadTasty.addTasty$$anonfun$1(ReadTasty.scala:33)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error]         at dotty.tools.dotc.core.Phases$Phase.monitor(Phases.scala:510)
[error]         at dotty.tools.dotc.fromtasty.ReadTasty.addTasty(ReadTasty.scala:33)
[error]         at dotty.tools.dotc.fromtasty.ReadTasty.runOn$$anonfun$1(ReadTasty.scala:28)
[error]         at scala.collection.Iterator$$anon$6.hasNext(Iterator.scala:479)
[error]         at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:619)
[error]         at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:617)
[error]         at scala.collection.AbstractIterable.foreach(Iterable.scala:935)
[error]         at scala.collection.IterableOps$WithFilter.foreach(Iterable.scala:905)
[error]         at dotty.tools.dotc.fromtasty.ReadTasty.runOn(ReadTasty.scala:28)
[error]         at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:380)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error]         at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1324)
[error]         at dotty.tools.dotc.Run.runPhases$1(Run.scala:373)
[error]         at dotty.tools.dotc.Run.compileUnits$$anonfun$1$$anonfun$2(Run.scala:420)
[error]         at dotty.tools.dotc.Run.compileUnits$$anonfun$1$$anonfun$adapted$1(Run.scala:420)
[error]         at scala.Function0.apply$mcV$sp(Function0.scala:42)
[error]         at dotty.tools.dotc.Run.showProgress(Run.scala:482)
[error]         at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:420)
[error]         at dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:432)
[error]         at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:69)
[error]         at dotty.tools.dotc.Run.compileUnits(Run.scala:432)
[error]         at dotty.tools.dotc.Run.compileUnits(Run.scala:325)
[error]         at dotty.tools.dotc.fromtasty.TASTYRun.compile(TASTYRun.scala:13)
[error]         at dotty.tools.dotc.Driver.doCompile(Driver.scala:37)
[error]         at dotty.tools.dotc.Driver.process(Driver.scala:201)
[error]         at scala.tasty.inspector.ScaladocInternalTastyInspector$.inspectAllTastyFilesInContext(TastyInspector.scala:67)
[error]         at dotty.tools.scaladoc.tasty.ScaladocTastyInspector$.loadDocs(TastyParser.scala:163)
[error]         at dotty.tools.scaladoc.ScalaModuleProvider$.mkModule(ScalaModuleProvider.scala:11)
[error]         at dotty.tools.scaladoc.Scaladoc$.run(Scaladoc.scala:242)
[error]         at dotty.tools.scaladoc.Scaladoc$.run$$anonfun$1(Scaladoc.scala:71)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error]         at scala.Option.map(Option.scala:242)
[error]         at dotty.tools.scaladoc.Scaladoc$.run(Scaladoc.scala:56)
[error]         at dotty.tools.dottydoc.Main$.process(Main.scala:25)
[error]         at dotty.tools.dottydoc.Main.process(Main.scala)
[error]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[error]         at xsbt.DottydocRunner.run(DottydocRunner.java:61)
[error]         at xsbt.ScaladocInterface.run(ScaladocInterface.java:11)
[error]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[error]         at sbt.internal.inc.AnalyzingCompiler.invoke(AnalyzingCompiler.scala:329)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:175)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:133)
[error]         at sbt.Doc$.$anonfun$scaladoc$1(Doc.scala:53)
[error]         at sbt.Doc$.$anonfun$scaladoc$1$adapted(Doc.scala:41)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1(RawCompileLike.scala:80)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1$adapted(RawCompileLike.scala:73)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4(RawCompileLike.scala:64)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4$adapted(RawCompileLike.scala:62)
[error]         at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:220)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1(RawCompileLike.scala:69)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1$adapted(RawCompileLike.scala:53)
[error]         at sbt.Defaults$.$anonfun$docTaskSettings$4(Defaults.scala:2276)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error]         at sbt.Execute.work(Execute.scala:292)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
[error]         at java.base/java.lang.Thread.run(Thread.java:1583)
[error] (scala-library-nonbootstrapped / Compile / doc) java.lang.reflect.InvocationTargetException
```
</details>

--- a/.github/workflows/stdlib.yaml
+++ b/.github/workflows/stdlib.yaml
@@ -722,3 +722,25 @@ jobs:
       - name: Run Community Build C
         run: |
           ./project/scripts/sbt "community-build/testOnly dotty.communitybuild.CommunityBuildTestC"
+
+  #################################################################################################
+  ######################################## GENERATE DOCS ##########################################
+  #################################################################################################
+
+  scala-library-docs:
+    runs-on: ubuntu-latest
+    needs  : [scala-library-bootstrapped, scaladoc]
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v6
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v5
+        with:
+          distribution: 'temurin'
+          java-version: 17
+          cache: 'sbt'
+
+      - uses: sbt/setup-sbt@v1
+      - name: Generate Documentation of the Standard Library
+        run: ./project/scripts/sbt scala-library-bootstrapped/doc

--- a/project/Build.scala
+++ b/project/Build.scala
@@ -1239,8 +1239,8 @@ object Build {
       // Add the source directories for the stdlib (non-boostrapped)
       Compile / unmanagedSourceDirectories := Seq(baseDirectory.value / "src"),
       Compile / unmanagedSourceDirectories += baseDirectory.value / "src-non-bootstrapped",
-      Compile / scalacOptions += "-Yno-stdlib-patches",
-      Compile / scalacOptions ++= Seq(
+      Compile / compile / scalacOptions += "-Yno-stdlib-patches",
+      Compile / compile / scalacOptions ++= Seq(
         // Needed so that the library sources are visible when `dotty.tools.dotc.core.Definitions#init` is called
         "-sourcepath", (Compile / sourceDirectories).value.map(_.getCanonicalPath).distinct.mkString(File.pathSeparator),
       ),
@@ -1312,7 +1312,6 @@ object Build {
   lazy val `scala-library-bootstrapped` = project.in(file("library"))
     .enablePlugins(ScalaLibraryPlugin)
     .settings(publishSettings)
-    .settings(disableDocSetting) // TODO now produces empty JAR to satisfy Sonatype, see https://github.com/scala/scala3/issues/24434
     .settings(
       name          := "scala-library-bootstrapped",
       moduleName    := "scala-library",
@@ -1331,8 +1330,8 @@ object Build {
       // Add the source directories for the stdlib (non-boostrapped)
       Compile / unmanagedSourceDirectories := Seq(baseDirectory.value / "src"),
       Compile / unmanagedSourceDirectories += baseDirectory.value / "src-bootstrapped",
-      Compile / scalacOptions += "-Yno-stdlib-patches",
-      Compile / scalacOptions ++= Seq(
+      Compile / compile / scalacOptions += "-Yno-stdlib-patches",
+      Compile / compile / scalacOptions ++= Seq(
         // Needed so that the library sources are visible when `dotty.tools.dotc.core.Definitions#init` is called
         "-sourcepath", (Compile / sourceDirectories).value.map(_.getCanonicalPath).distinct.mkString(File.pathSeparator),
       ),
@@ -1465,9 +1464,9 @@ object Build {
       Compile / unmanagedSourceDirectories := Seq(baseDirectory.value / "src"),
       Compile / unmanagedSourceDirectories ++=
         (`scala-library-bootstrapped` / Compile / unmanagedSourceDirectories).value,
-      Compile / scalacOptions += "-Yno-stdlib-patches",
+      Compile / compile / scalacOptions += "-Yno-stdlib-patches",
       // Configure the source maps to point to GitHub for releases
-      Compile / scalacOptions ++= {
+      Compile / compile / scalacOptions ++= {
         if (isRelease) {
           val baseURI = (LocalRootProject / baseDirectory).value.toURI
           val dottyVersion = version.value

